src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -112,24 +135,51 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     return ETH_NETWORK_DEFAULT_SETTINGS[networkType];
   }
 
-  override async fetchBalance() {
-    return ''; // TODO
+  override async fetchBalance(address: string): Promise<string> {
+    const { ethActiveNetworkSettings } = useEthNetworkSettings();
+    const { nodeUrl } = ethActiveNetworkSettings.value;
+    const web3Eth = new Web3Eth(nodeUrl);
+    const balanceInWei = await getBalance(web3Eth, address, 'latest', NUMBER_DATA_FORMAT);
+    return fromWei(balanceInWei, 'ether').toString();
   }
 
   override isAccountAddressValid(address: string) {
     return isAddress(address);
   }
 
-  override async isAccountUsed() {
-    return false; // TODO
+  override async isAccountUsed(address: string): Promise<boolean> {
+    const balance = parseFloat(await this.fetchBalance(address));
+    if (balance > 0) {
+      return true;
+    }
+    // in case of 0 balance, check if account has transaction history
+    const txCount = await this.getTransactionCount(address);
+    return txCount > 0;
   }
 
-  override getHdWalletAccountFromMnemonicSeed(): IHdWalletAccount {
-    return {} as any; // TODO
+  override getHdWalletAccountFromMnemonicSeed(
+    seed: Uint8Array,
+    accountIndex: number,
+  ): IHdWalletAccount {
+    const hdNodeWallet = this.bip32.fromSeed(Buffer.from(seed));
+    const path = `m/44'/60'/0'/0/${accountIndex}`;
```

Comment:
This derivation path is incorrect, you are deriving an address, and not a new account.